client/client.go

```diff
@@ -183,9 +183,9 @@ func (cli *Client) Close() error {
 
 // getAPIPath returns the versioned request path to call the api.
 // It appends the query parameters to the path if they are not empty.
-func (cli *Client) getAPIPath(p string, query url.Values) string {
+func (cli *Client) getAPIPath(p string, query url.Values, ignoreVersion bool) string {
```

Comment:
@vieux this `client/request.go` has a few issues I think. I've been meaning to refactor these methods for a while now. I think the problem we're seeing here is a symptom of those issues.

I refactored the send methods here: https://github.com/docker/docker/compare/master...dnephin:refactor-client

That branch includes a commit which uses the refactored functions to call `Ping()` without the api version in the path. It removes the need to have the `ignoreVersion` parameter entirely.

If this works for you, would you like to cherry pick the first commit into this branch? Otherwise I can drop the change to `ping.go` and open it as a PR.  Let me know which you prefer.